features/dd-sdk-android-rum/src/main/kotlin/com/datadog/android/rum/RumMonitor.kt

```diff
@@ -126,7 +126,7 @@ interface RumMonitor {
             " Use `startResource` method which takes `RumHttpMethod` as `method` parameter instead."
     )
     fun startResource(
-        key: String,
+        key: Any,
```

Comment:
I wonder why we need to keep `Any` type here? can we just use the new class `ResourceId` instead?